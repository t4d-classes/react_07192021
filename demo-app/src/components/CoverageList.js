import PropTypes from 'prop-types';

export const CoverageList = ({ children, coverages }) => {
  return (
    <>
      <header>
        {children}
      </header>
      <ul>
        {coverages.map(c =>
          <li key={c.id}>
            {c.name}
          </li>)}
      </ul>
    </>
  );
};

CoverageList.defaultProps = {
  coverages: []
};

CoverageList.propTypes = {
  coverages: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  })),
  children: PropTypes.node.isRequired,
};

