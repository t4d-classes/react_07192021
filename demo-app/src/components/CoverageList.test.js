import React from "react";

import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import { CoverageList } from "./CoverageList";

test("snapshot CoverageList component", () => {

  const coverages = [
    { id: 1, name: "First" },
    { id: 2, name: "Second" },
    { id: 3, name: "Third" },
  ];

  expect(
    renderer.create(
      <CoverageList coverages={coverages}>
        Some Content
      </CoverageList>
    ).toJSON()
  ).toMatchSnapshot();
});

describe("CoverageList component", () => {

  let coverages;

  beforeEach(() => {
    coverages = [
      { id: 1, name: "First" },
      { id: 2, name: "Second" },
      { id: 3, name: "Third" },
    ];
  });

  test("renders CoverageList component", () => {
    const { getByRole, getAllByRole } = render(
      <CoverageList coverages={coverages}>
        <h1>List of Coverages</h1>
      </CoverageList>
    );


    expect(getByRole("banner").innerHTML)
      .toBe("<h1>List of Coverages</h1>");

    const listItems = getAllByRole("listitem");

    expect(listItems.length).toBe(3);

    coverages.forEach((coverage, index) => {
      expect(listItems[index].textContent).toBe(coverage.name);
    });

  });
});
