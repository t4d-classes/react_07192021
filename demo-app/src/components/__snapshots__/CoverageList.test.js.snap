// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot CoverageList component 1`] = `
Array [
  <header>
    Some Content
  </header>,
  <ul />,
]
`;
